### PR TITLE
fix: WynntilsAccountManager's token users are now async

### LINKED ISSUE #829: Every consumer of Wynntils Account Tokens should be able to run asynchronously
See https://github.com/Wynntils/Artemis/pull/826#discussion_r1048942877.

### PR DESCRIPTION
**Pull request by kristofbolyai**
Fixes #829

### CODE REVIEW

File: common/src/main/java/com/wynntils/sockets/model/HadesModel.java [hunk 1]
```diff
@@ -47,20 +48,25 @@ public final class HadesModel extends Model {
...
+            tryCreateConnection();
+        }
     }
 
     @Override
     public void disable() {
         tryDisconnect();
     }
 
-    private void tryCreateConnection() {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/sockets/model/HadesModel.java:58]: Most of the PR looks good, but this one kind of bugged me. If Athena login works but Hades connection fails, we will not get any feedback on this. Just like the Athena firstWorld message, maybe we need to do something similar for Hades?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/sockets/model/HadesModel.java:58]: Fixed

File: common/src/main/java/com/wynntils/sockets/model/HadesModel.java [hunk 2]
```diff
@@ -127,6 +135,22 @@ public void onPartyListUpdate(RelationsUpdateEvent.PartyList event) {
 
     @SubscribeEvent
     public void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldStateManager.State.WORLD && isSocketOpen() && firstWorldJoin) {
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: How does this check for failure?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: isSocketOpen, socket is closed on auth fail.

File: common/src/main/java/com/wynntils/sockets/model/HadesModel.java [hunk 3]
```diff
@@ -127,6 +135,22 @@ public void onPartyListUpdate(RelationsUpdateEvent.PartyList event) {
...
+            firstWorldJoin = false;
+
+            // FIXME: Use the proper reload command here, once they are reworked
+            MutableComponent failed = Component.literal("Welps! Trying to connect to Hades failed.")
+                    .withStyle(ChatFormatting.GREEN);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: These literals should really be translatable components. Also "Welps" is a) informal and b) probably should be welp, don't know what I was doing when I wrote that.

Comment (magicus) [reply common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: We can fix that now or later. I think we need to do a full sweep of the entire Artemis code base and fix all non-i18n strings. (The one exception is in log messages; they don't need to be i18n)

File: common/src/main/java/com/wynntils/sockets/model/HadesModel.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: But this looks wrong? Don't you first need to do an if check if this is firstWorldJoin, and set the field, and then check socket? Also, the socket check looks inverted.

**Review by magicus — APPROVED** [review]

File: common/src/main/java/com/wynntils/sockets/model/HadesModel.java (hunk 3 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/sockets/model/HadesModel.java]: I opened https://github.com/Wynntils/Artemis/issues/906 for this.

**Pull request merged**